pyhive/common.py

```diff
@@ -237,6 +247,10 @@ def escape_item(self, item):
             return self.escape_string(item)
         elif isinstance(item, collections.Iterable):
             return self.escape_sequence(item)
+        elif isinstance(item, datetime.datetime):
+            return self.escape_datetime(item, ParamEscaper._DATETIME_FORMAT)
+        elif isinstance(item, datetime.date):
+            return self.escape_datetime(item, ParamEscaper._DATE_FORMAT)
```

Comment:
I'll bow to @bkyryliuk 's opinion on implementation, but why are you re-instantiating the `ParamEscaper` object here, instead of just referencing `self`?

Reply:
I'm referencing `ParamEscaper._DATE_FORMAT` in the sense that these are 'constants' on the class and not specific/individual properties on the instance. I assume that Python is smart enough to not really instantiate a new object, but that this variable is just available from a singleton?

That's my reasoning at least :-) but I guess I could change it to `self._DATE_FORMAT` too.

Reply:
While technically true in this situation, I would recommend switching to `self`. Its the proper way to do it in Python. 

While it isn't technically an issue here, should the design and behavior of this class change in the future, you could find yourself referencing a variable that behaves differently when not referenced through the current instantiated version of the object. Plus with the naming convention you used, it is documented that those variables should only be accessed internally as they are considered 'Private'. https://docs.python.org/3/tutorial/classes.html#private-variables